morphman/automated_landmarking/automated_landmarking_piccinelli.py

```diff
@@ -0,0 +1,154 @@
+        raise ValueError("ERROR: Selected method for computing curvature / torsion not available" +
+                         "\nPlease select between 'spline' and 'vmtk'")
+
+    # Extract local curvature minimums
+    length = get_curvilinear_coordinate(line)
+
+    # Ignore points close to the ends of the siphon
+    for i in max_point_ids:
+        if length[i] in length[-10:] or length[i] in length[:10]:
+            max_point_ids.remove(i)
+
+    # Remove curvature and torsion peaks too close to each other
+    tolerance = 70
+    dist = []
+    dist_tor = []
+    for i in range(len(max_point_ids) - 1):
+        dist.append(max_point_ids[i + 1] - max_point_ids[i])
+    for i in range(len(max_point_tor_ids) - 1):
+        dist_tor.append(max_point_tor_ids[i + 1] - max_point_tor_ids[i])
+
+    curv_remove_ids = []
+    for i, dx in enumerate(dist):
+        if dx < tolerance:
+            curv1 = curvature[max_point_ids[i]]
+            curv2 = curvature[max_point_ids[i + 1]]
+            if curv1 > curv2:
+                curv_remove_ids.append(max_point_ids[i + 1])
+            else:
+                curv_remove_ids.append(max_point_ids[i])
+
+    tor_remove_ids = []
+    for i, dx in enumerate(dist_tor):
+        if dx < tolerance:
+            tor1 = torsion_smooth[max_point_tor_ids[i]]
+            tor2 = torsion_smooth[max_point_tor_ids[i + 1]]
+            if tor1 > tor2:
+                tor_remove_ids.append(max_point_tor_ids[i + 1])
+            else:
+                tor_remove_ids.append(max_point_tor_ids[i])
+
+    max_point_ids = [ID for ID in max_point_ids if ID not in curv_remove_ids]
+    max_point_tor_ids = [ID for ID in max_point_tor_ids if ID not in tor_remove_ids]
+
+    # Define bend interfaces based on Piccinelli et al.
+    def find_interface():
+        found = False
+        interface = {}
+        k = 0
+        start_id = 0
+        for c in max_point_ids:
+            for i in range(start_id, len(max_point_tor_ids) - 1):
+                if max_point_tor_ids[i] < c < max_point_tor_ids[i + 1] and not found:
+                    interface["bend%s" % (k + 1)] = np.array([max_point_tor_ids[i]])
+                    k += 1
+                    interface["bend%s" % (k + 1)] = np.array([max_point_tor_ids[i + 1]])
+                    k += 1
+                    start_id = i + 1
+                    found = True
+            found = False
+
+        return interface
+
+    # Compute and extract interface points
+    interfaces = find_interface()
+    landmarks = {}
+    for k, v in interfaces.items():
+        landmarks[k] = line.GetPoints().GetPoint(int(v))
+
+    # Map landmarks to initial centerline
+    landmarks = map_landmarks(landmarks, centerline, algorithm)
+
+    # Save landmarks
+    print("-- Case was successfully landmarked.")
+    print("-- Number of landmarks (Segments): %s" % len(landmarks))
+    try:
+        os.remove(base_path + "_landmark_piccinelli_%s.particles" % approximation_method)
+    except:
```

Comment:
Please add the specific error you would like to catch, I guess `FileNotFoundError` or `PermissionError` is the relevant ones here.